src/osd/PGLog.cc

```diff
@@ -399,15 +438,93 @@ void PGLog::merge_log(pg_info_t &oinfo, pg_log_t &olog, pg_shard_t fromosd,
 
     changed = true;
   }
-  
-  dout(10) << "merge_log result " << log << " " << missing << " changed=" << changed << dendl;
+
+  // now handle dups
+  merge_log_dups(olog, changed);
```

Comment:
should set dirty_dups here if (changed), so they are persisted when write_log_and_missing() is called